R/expectation.R

```diff
@@ -260,11 +260,14 @@ expectation_location <- function(x) {
   if (is.null(x$srcref)) {
     "???"
   } else {
-    filename <- attr(x$srcref, "srcfile")$filename
+    srcfile <- attr(x$srcref, "srcfile")
+    filename <- srcfile$filename
     if (identical(filename, "")) {
       paste0("Line ", x$srcref[1])
     } else {
-      paste0(basename(filename), ":", x$srcref[1], ":", x$srcref[2])
+      loc <- paste0(basename(filename), ":", x$srcref[1], ":", x$srcref[2])
+      filepath <- paste0("file::", file.path(srcfile$wd, filename), "#", x$srcref[1], ",", x$srcref[2])
+      cli::style_hyperlink(loc, filepath)
```

Comment:
Not sure this is the only place that needs a link.

Reply:
I'm reasonably confident that all display of file names + line numbers now goes through this file, so it should be ok.